tools/font-edit/README.md

```diff
@@ -14,4 +18,66 @@ make
 ./twin-fedit < nchars
 ```
 
-(press 'q' to next character)
+### Background
+`twin-fedit` uses a singly linked list to store drawing-related operations. The nodes of this list are represented by the `cmd_t` structure, where each structure contains an operation type `op_t` and points `pt_t`. 
+
+```c
+typedef struct _cmd {
+    struct _cmd *next;
+    op_t op;
+    pt_t pt[3];
+} cmd_t;
+```
+The operation `op` determines the number of values in `pt`. For example, when `op_t = op_move`, `pt[0].x = 10` and `pt[0].y = 4` represent moving a point to (10, 4). When `op_t = op_curve`, `pt[0]` and `pt[1]` are the control points of a cubic Bézier curve, and `pt[2]` is the endpoint, used to draw the curve.
+
+If `op` is `op_move` or `op_line`, then `pt[1]` and `pt[2]` are undefined.
+
+The `cmd_stack_t` is a singly linked list that stores multiple `cmd_t` singly linked lists in a stack-like manner. The purpose of cmd_stack is to record all operations. To revert to the previous operation, this can be achieved using the stack.
+
+The `char_t` structure is used to store the `cmd_t` structures needed for drawing characters.
+```c
+typedef struct _char_t {
+    cmd_t *cmd;
+    cmd_stack_t *stack;
+    cmd_t *first;
+    cmd_t *last;
+} char_t;
+```
+
+### Key bindings
+
+| Key  | Functionality |
+|   ---       |   ---    |
+|  ESC  | Exit program |
+| left mouse button | Select a point and its corresponding `cmd_t` as the first pointer |
```

Comment:
Don't mention the identifiers in source code. Instead, illustrate with high-level concepts.